### PR TITLE
Support for re-registering extensions and modules (FATE#318800)

### PR DESCRIPTION
**Pull request by lslezak**
Note: There is still one part missing: the old subscription should be "unlinked" before registering an extension again. But this requires an extension in the `connect` gem which won't be ready in this sprint, I'll add it later when it's ready.

### CODE REVIEW

**Comment by mchf** [comment]
Travis fails

File: src/lib/registration/ui/addon_selection_base_dialog.rb [hunk 1]
```diff
@@ -39,45 +31,33 @@ def initialize(registration)
...
-          # help text (3/3)
-          _("<p>If you want to remove any extension or module you need to log"\
-              "into the SUSE Customer Center and remove them manually there.</p>"),
-          # always enable Back/Next, the dialog cannot be the first in workflow
-          true,
-          true
-        )
-
-        @old_selection = Addon.selected.dup
+        raise "Not implented"
```
Comment (mchf) [inline src/lib/registration/ui/addon_selection_base_dialog.rb]: typo

**Comment by lslezak** [comment]
Travis is OK, just coveralls fails because of the 0.31% test coverage decrease...

File: src/lib/registration/ui/addon_selection_reregistration_dialog.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+          # help text (2/3)
+          _("<p>Please note, that some extensions or modules might need "\
+              "specific registration code.</p>") +
+          # help text (3/3)
+          _("<p>If you want to remove any extension or module you need to log"\
+              "into the SUSE Customer Center and remove them manually there.</p>"),
+          # always enable Back/Next, the dialog cannot be the first in workflow
+          true,
+          true
+        )
```
Comment (mchf) [inline src/lib/registration/ui/addon_selection_reregistration_dialog.rb]: texts are same as in the other class, may be it deserve an optimization.

**Comment by mchf** [comment]
LGTM for me

**Pull request merged**